### PR TITLE
feat: Versioning Api AM-381

### PR DESCRIPTION
**Pull request by abhijitdiwan**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/AM-381

Change description
feat: Versioning Api AM-381

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -69,6 +65,18 @@ withPipeline(type, product, component) {
   }
 
   onMaster {
+    secrets = [
+      (appVault): [
```
Comment (bmcts) [inline Jenkinsfile_CNP]: The secrets are already defined globally at the top, please remove this from here

File: am-lib/src/main/resources/db/migrationAks/V19__load_data.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+insert into services (service_name, service_description) values ('cmc-test', null)
+on conflict on constraint services_pkey do nothing;
```
Comment (bmcts) [inline am-lib/src/main/resources/db/migrationAks/V19__load_data.sql]: Weren't we trying not to put any values into the DB in the flyway scripts?

File: src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/AccessManangementApiTest.java [hunk 3]
```diff
@@ -41,7 +41,7 @@ public void verifyGrantExplicitAccessApi() {
             + "}";
         try {
             Response response = amApiClient.createExplicitAccess().post(amApiClient.getAccessUrl()
-                + "api/access-resource");
+                + "api/v1/access-resource");
```
Comment (bmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/amapi/functional/AccessManangementApiTest.java]: Please define a global "version" variable and replace all of the version hardcoding in this file

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -53,11 +53,7 @@ withPipeline(type, product, component) {
   env.S2S_URL = "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
 
   before('functionalTest:preview') {
-    env.URL_TO_TEST = "http://am-lib-test-aat.service.core-compute-aat.internal"
-  }
-
-  before('functionalTest:aat') {
-    env.URL_TO_TEST = env.TEST_URL
+    env.environment_name ="preview"
```
Comment (bmcts) [inline Jenkinsfile_CNP]: Is this required? The stage name already has "preview" as the environment

File: src/gatling/simulations/uk/gov/hmcts/reform/amlib/performance/http/AccessManagement.scala [hunk 5]
```diff
@@ -8,21 +8,21 @@ object AccessManagement {
 
   private def getRequest(url: String): HttpRequestBuilder =
     http("/returnResourceAccessors")
-      .get("/api" + url)
+      .get("/api/v1" + url)
```
Comment (bmcts) [inline src/gatling/simulations/uk/gov/hmcts/reform/amlib/performance/http/AccessManagement.scala]: Please define a global "version" variable and replace all of the version hardcoding in this file

File: src/integrationTest/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java [hunk 6]
```diff
@@ -99,7 +99,7 @@ public void testCreateResourceAccess() throws Exception {
 
         doNothing().when(accessManagementService).grantExplicitResourceAccess(any());
 
-        this.mockMvc.perform(post("/api/access-resource")
+        this.mockMvc.perform(post("/api/v1/access-resource")
```
Comment (bmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: Please define a global "version" variable and replace all of the version hardcoding in this file

File: src/integrationTest/java/uk/gov/hmcts/reform/amapi/exception/AccessManagementResponseEntityExceptionHandlerTest.java [hunk 7]
```diff
@@ -76,7 +76,7 @@ public void testHandleHttpMessageNotReadable() throws Exception {
         String invalidJson = Resources.toString(Resources
             .getResource("exception-mapper-data/malformedInput.json"), StandardCharsets.UTF_8);
 
-        this.mockMvc.perform(post("/api/filter-resource")
+        this.mockMvc.perform(post("/api/v1/filter-resource")
```
Comment (bmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/amapi/exception/AccessManagementResponseEntityExceptionHandlerTest.java]: Please define a global "version" variable and replace all of the version hardcoding in this file

File: src/main/resources/application.yaml [hunk 8]
```diff
@@ -26,8 +26,8 @@ spring:
     password: ${AM_DB_PASSWORD:ampass}
     properties:
       charSet: UTF-8
-  flyway:
-    enabled: ${ENABLE_DB_MIGRATE_IN_SERVICE:true}
+  flyway.locations: classpath:db/migration,classpath:db/migrationAks
+  flyway.enabled: ${ENABLE_DB_MIGRATE_IN_SERVICE:true}
```
Comment (bmcts) [inline src/main/resources/application.yaml]: Please change this to YAML syntax as per the rest of the file (ie. no full stops)

File: src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerUnitTest.java [hunk 9]
```diff
@@ -86,7 +86,7 @@ public void testCreateResourceAccess() throws Exception {
 
         doNothing().when(accessManagementService).grantExplicitResourceAccess(any());
 
-        this.mvc.perform(post("/api/access-resource")
+        this.mvc.perform(post("/api/v1/access-resource")
```
Comment (bmcts) [inline src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerUnitTest.java]: Please define a global "version" variable and replace all of the version hardcoding in this file

File: am-lib/src/main/resources/db/migrationAks/V19__load_data.sql (hunk 2 above)
Comment (abhijitdiwan) [reply am-lib/src/main/resources/db/migrationAks/V19__load_data.sql]: Yes..Fly way is only applicable for Local and AKS. And will not run in AAT as aat used Jenkins pipe line where we ar only using default migration location.

File: Jenkinsfile_CNP (hunk 4 above)
Comment (abhijitdiwan) [reply Jenkinsfile_CNP]: Yes accessing it while inserting data.

File: src/integrationTest/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java (hunk 6 above)
Comment (abhijitdiwan) [reply src/integrationTest/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: done

File: src/integrationTest/java/uk/gov/hmcts/reform/amapi/exception/AccessManagementResponseEntityExceptionHandlerTest.java (hunk 7 above)
Comment (abhijitdiwan) [reply src/integrationTest/java/uk/gov/hmcts/reform/amapi/exception/AccessManagementResponseEntityExceptionHandlerTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerUnitTest.java (hunk 9 above)
Comment (abhijitdiwan) [reply src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerUnitTest.java]: done

**Pull request merged**